src/Orleans/CodeGeneration/KnownTypeAttribute.cs

```diff
@@ -2,17 +2,34 @@ namespace Orleans.CodeGeneration
 {
     using System;
 
-    /// <summary>
-    /// The attribute which informs the code generator that code should be generated a type.
-    /// </summary>
-    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
-    public class KnownTypeAttribute : Attribute
+    public class ConsiderForCodeGenerationAttribute : Attribute
     {
-        public KnownTypeAttribute(Type type)
+        protected ConsiderForCodeGenerationAttribute(Type type, bool generateSerializer)
         {
             this.Type = type;
+            this.GenerateSerializer = generateSerializer;
         }
 
-        public Type Type { get; set; }
+        public Type Type { get; private set; }
+        public bool GenerateSerializer { get; private set; }
```

Comment:
I know that a user can just use the derived GenerateSerializer attribute, but is there a reason not to make the setter public too? I'm fine either way, I just found it a little odd, but not opposed anyway.